### DISCUSSION TITLE
Future validation of password

### DISCUSSION THREAD

**Discussion by caouecs** [Laravel-Lang/common#39] [Q&A]
https://github.com/laravel/framework/pull/36960

with new tags, but not defined for the moment

**Comment by andrey-helldar** [comment]
I now have one less package to support ([andrey-helldar/strong-password](https://github.com/andrey-helldar/strong-password)) 😅

**Comment by caouecs** [comment]
the pull request was merger but without translations...

**Reply by luisprmat** [reply]
@caouecs, No, [laravel/framework#36960](https://github.com/laravel/framework/pull/36960) was merged on April 23 and it works perfectly with translations. i tested it with Spanish Language.

it happens that this time the *messages* are wrapped by the translator through the **fail** method of the `Illuminate\Validation\Rules\Password` class.

```php
    $this->fail('The :attribute must contain at least one uppercase and one lowercase letter.');
...
    protected function fail($messages)
    {
        $messages = collect(Arr::wrap($messages))->map(function ($message) {
            return __($message);  // Call translator
        })->all();

        $this->messages = array_merge($this->messages, $messages);

        return false;
    }
```

I tried this code in the **register form** provided by [**Laravel Breeze**](https://laravel.com/docs/8.x/starter-kits#laravel-breeze), modifying `app/Http/Controllers/Auth/RegisteredUserController.php` in the **store** method.

```php
use Illuminate\Validation\Rules\Password;

...

$request->validate([
    'name' => 'required|string|max:255',
    'email' => 'required|string|email|max:255|unique:users',
    'password' => [
        'required',
        'string',
        'confirmed',
        Password::min(2)
            ->symbols()
            ->numbers()
            ->mixedCase()
            ->letters()
            ->uncompromised()
    ],
]);
```
and i got 

![es_trans_01](https://user-images.githubusercontent.com/9275870/115979177-0d0c3980-a54a-11eb-80e3-dc7027f5496d.PNG)

This feature yet doesn't available in Laravel **8.38.0** but i tested in *dev mode* (branch **8.x-dev**). I guess it will be available in the next release (**8.39.0** ?).
For this reason I added two **PR** ( [en](https://github.com/Laravel-Lang/lang/pull/1717), [es](https://github.com/Laravel-Lang/lang/pull/1718)) since the tests do not pass after modifying the file `source/en.json` and it did not allow me to make a single pull request from my fork as I always do.

Hopefully @andrey-helldar can fix this soon.

**Reply by andrey-helldar** [reply]
I remembered why the test was giving an error.

You added new keys to the `source/en.json` file, but did not add them to all languages.

In this case, we just accept the PR without paying attention to phpunit.

When the PR is accepted, an automatic script is launched that will put down all the necessary keys (`php app/keys.php`)

**Comment by luisprmat** [comment]
> [laravel/framework#36960](https://github.com/laravel/framework/pull/36960)
> 
> with new tags, but not defined for the moment

Feature supported since version **8.39.0** which was released today April 27
